src/libs/karm-base/ring.h

```diff
@@ -69,7 +70,7 @@ struct Ring {
             panic("pop on empty ring");
 
         T value = _buf[_head].take();
-        _head = (_head - 1) % _cap;
+        _head = (_head + (_cap - 1)) % _cap;
```

Comment:
doing `head-1` could go to usize max if head=0 which won't have the correct mod. I didn't do the math but pretty sure it would happen, mostly if _cap is not a divisor of usize capacity